chore: update concepts (#1070)
zama-ai/concrete-ml: docs/llm/inference.md

# Encrypted LLM Inference

LLMs can be converted to use FHE to generate encrypted tokens based on encrypted prompts. Concrete ML implements LLM inference as a client/server protocol where

> - the client executes non-linear layers in the LLM, such as attention and activation functions
> - the server executes linear layers, such as projection and embedding

Reviewer: ```suggestion
- The client executes non-linear layers in the LLM, such as attention and activation functions.
- The server executes linear layers, such as projection and embedding.
```

The FHE LLM implementation in Concrete ML has the following characteristics:

> - data transfer is necessary for each linear layer. The size of encrypted data
>   is about 4x the size of the clear data that are input/outputs to the linear layers. For a [LLAMA 1B](https://huggingface.co/meta-llama/Llama-3.2-1B) model, around 18MB of data are exchanged per token, while for a \[GPT2\] model around 2.2MB are exchanged.
> - the client machine needs to perform some computation, thus it needs to execute some PyTorch layers
> - the advantage of using FHE resides in outsourcing computation when the client
>   has low hardware capabilities, or when the model IP needs to be protected

Reviewer: ```suggestion
- Data transfer is necessary for each linear layer. The size of encrypted data
  is about 4x the size of the clear data that are input/outputs to the linear layers. For instance:
   - A [LLAMA 1B](https://huggingface.co/meta-llama/Llama-3.2-1B) model exchanges around 18MB of data per token.
   - A \[GPT2\] mode exchanges around 2.2MB of data per token.
- The client machine needs to perform some computation, thus it needs to execute some PyTorch layers.
- Advantages of FHE include:
  - Offloading computation from clients with limited hardware.
  - Preserving intellectual property by running sensitive model components on encrypted data.
```

## Compiling an LLM for FHE Inference

> 

Reviewer: ```suggestion
This document introduces how to use Concrete ML to run encrypted LLM inference with FHE.
```

To prepare an LLM model for FHE inference, use the `HybridFHEModel` class:

> ```python

Reviewer 2: is it a tested code block? I don't remind the syntax

Author: yes, it is!

Author: we'll add a GPU CI soon maybe we can only run this test on the GPU CI
